app/component/history/history.html

```diff
@@ -2,7 +2,7 @@
   <h2> history </h2>
   <button ng-click="historyCtrl.visitor.undo()"> undo </button>
   <ol>
-    <li ng-repeat="item in historyCtrl.visitor.history track by item.id">
+    <li ng-repeat="item in historyCtrl.visitor.history track by $index">
       <p> {{ item.desc }} </p>
       <button ng-click="historyCtrl.visitor.removeMove(item.id)"> del </button>
```

Comment:
Your ``del`` button doesn't do anything because you didn't make a ``removeItem`` function in your ``visitor-service.``